Orthtree/include/CGAL/Orthtree.h

```diff
@@ -39,342 +41,332 @@
+  {
+    return m_node_contents[n];
+  }
+
+  Node_data& operator[](Node_index n)
+  {
+    return m_node_contents[n];
+  }
+};
+
+template <class GT>
+struct Node_data_wrapper<GT, false>
+{
+  using Node_index = typename GT::Node_index;
+  using Node_data = void*;
+
+  template <class Property_container>
+  Node_data_wrapper(Property_container&) {}
+
+  void* operator[](Node_index) const
+  {
+    return nullptr;
+  }
+};
+
+} // end of Orthtree_impl namespace
+
 /*!
-  \ingroup PkgOrthtreeClasses
+  \ingroup PkgOrthtreeRef
 
-  \brief A data structure using an axis-aligned hybercubic
-  decomposition of dD space for efficient point access and
-  computations.
+  \brief A data structure using an axis-aligned hyperrectangle
+  decomposition of dD space for efficient access and
+  computation.
 
-  \details It builds a hierarchy of nodes which subdivide the space
-  based on a collection of points.  Each node represents an
-  axis-aligned hypercubic region of space.  A node contains the range
-  of points that are present in the region it defines, and it may
+  \details It builds a hierarchy of nodes which subdivides the space.
+  Each node represents an axis-aligned hyperrectangle region of space.
+  The contents of nodes depend on the traits class, non-leaf nodes also
   contain \f$2^{dim}\f$ other nodes which further subdivide the
   region.
 
   \sa `CGAL::Quadtree`
   \sa `CGAL::Octree`
 
-  \tparam Traits_ must be a model of `OrthtreeTraits`
-  \tparam PointRange_ must be a model of range whose value type is the key type of `PointMap_`
-  \tparam PointMap_ must be a model of `ReadablePropertyMap` whose value type is `Traits_::Point_d`
+  \tparam GeomTraits must be a model of `OrthtreeTraits` or `OrthtreeTraitswithData`.
  */
-template<typename Traits_, typename PointRange_,
-         typename PointMap_ = Identity_property_map<typename Traits_::Point_d> >
-class Orthtree
-{
-
+template <typename GeomTraits>
+class Orthtree {
 public:
-
   /// \name Template Types
   /// @{
-  typedef Traits_ Traits; ///< Geometry traits
-  typedef PointRange_ PointRange; ///< Point range
-  typedef PointMap_ PointMap; ///< Point map
+  using Traits = GeomTraits; ///< Geometry traits
   /// @}
 
   /// \name Traits Types
   /// @{
-  typedef typename Traits::Dimension Dimension; ///< Dimension of the tree
-  typedef typename Traits::FT FT; ///< Number type.
-  typedef typename Traits::Point_d Point; ///< Point type.
-  typedef typename Traits::Bbox_d Bbox; ///< Bounding box type.
-  typedef typename Traits::Sphere_d Sphere; ///< Sphere type.
+#ifndef DOXYGEN_RUNNING
+  static inline constexpr bool has_data = Orthtree_impl::has_Node_data<GeomTraits>::value;
+  static inline constexpr bool supports_neighbor_search = true;// Orthtree_impl::has_Squared_distance_of_element<GeomTraits>::value;
+#else
+  static inline constexpr bool has_data = bool_value; ///< `true` if `GeomTraits` is a model of `OrthtreeTraitswithData` and `false` otherwise.
```

Comment:
```suggestion
  static inline constexpr bool has_data = bool_value; ///< `true` if `GeomTraits` is a model of `OrthtreeTraitsWithData` and `false` otherwise.
```